### ISSUE TITLE
sf::Window:setIcon causes segfault on Linux

### ISSUE THREAD

**Issue by mstraube** [SFML/SFML#1168]
Using sf::Window::setIcon leads to a segfault on Linux. This [example](https://gist.github.com/mstraube/a557ee06660fc95aa7a374f641fc27ce) gives following gdb output:
```
Reading symbols from ./sfml...done.
(gdb) r
Starting program: /home/john/sfml/sfml 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fffecdb4700 (LWP 28918)]
[New Thread 0x7fffec3ae700 (LWP 28919)]
[New Thread 0x7fffebbad700 (LWP 28920)]
[New Thread 0x7fffeb3ac700 (LWP 28921)]
[New Thread 0x7fffeabab700 (LWP 28922)]
Success loading icon.png

Thread 1 "sfml" received signal SIGSEGV, Segmentation fault.
0x00007ffff6082c50 in _XData32 () from /usr/lib/libX11.so.6
(gdb) bt
#0  0x00007ffff6082c50 in _XData32 () from /usr/lib/libX11.so.6
#1  0x00007ffff605d683 in XChangeProperty () from /usr/lib/libX11.so.6
#2  0x00007ffff79752da in sf::priv::WindowImplX11::setIcon (this=0x7f3a40, width=256, height=256, 
    pixels=<optimized out>) at /build/sfml/src/SFML/src/SFML/Window/Unix/WindowImplX11.cpp:859
#3  0x000000000040151d in main () at sfml.cpp:20
(gdb) q
```
Seems 9996b7abb6c6e216dc40d527b0339c89880709c5 is the problem. I tested the last revision before that commit and it worked. Problem is also present in latest revision 746bb9c8cab04d44774ea1338c14f903cbaff3ae

I ran into this on x86_64 Arch Linux installation with sfml-2.4.1-1

**Comment by binary1248** [comment]
Should be fixed in #1171. @mstraube care to test?

**Comment by mstraube** [comment]
Yes, with https://github.com/SFML/SFML/pull/1171 it works.

**Comment by eXpl0it3r** [comment]
Fixed with #1171
